tests/test_ada_url.py

```diff
@@ -394,3 +398,31 @@ def test_idna_encode(self):
             idna_to_ascii('meßagefactory.ca'.encode('utf-8')),
             b'xn--meagefactory-m9a.ca',
         )
+
+
+class ParseTests(TestCase):
+    def test_url_suite(self):
+        with open(URL_TEST_DATA_PATH, 'rb') as f:
+            test_data = load(f)
+
+        for i, item in enumerate(test_data, 1):
+            # Skip the comments
+            if isinstance(item, str):
+                continue
+
+            # Skip tests that can't be represented properly with the json module
+            try:
+                (item.get('input') or '').encode('utf-8')
+                (item.get('base') or '').encode('utf-8')
```

Comment:
base can be "null" as a string.

Reply:
Per the [readme](https://github.com/web-platform-tests/wpt/tree/master/url#urltestdatajson):

```
The format of resources/urltestdata.json is a JSON array of comments as strings and test cases as objects. The keys for each test case are:

    input: a string to be parsed as URL.

    base: null or a serialized URL (i.e., does not fail parsing).

```

Surely this means the `"null"` the string is meant to be interpreted as a string, right?

---

In any case, this test is simply to identify test case 410, the one with the questionably-encoded-in-JSON Unicode surrogates. The actual parse is done below.

Reply:
My mistake!